From akpm@osdl.org Tue Mar 28 15:22:02 2006
Message-Id: <200603282321.k2SNLg5T000638@shell0.pdx.osdl.net>
Subject: [patch 1/1] hid-core.c: fix "input irq status -32 received" for Silvercrest USB Keyboard
To: greg@kroah.com
Cc: akpm@osdl.org, jeffrey@wirehead.be, david-b@pacbell.net, stern@rowland.harvard.edu
From: akpm@osdl.org
Date: Tue, 28 Mar 2006 15:21:36 -0800


From: Jeffrey Vandenbroucke sign <jeffrey@wirehead.be>

When not using this patch, the kernel will continuously return "input irq
status -32 received", while making the keyboard unusable.  This can be
easely resolved using HID_QUIRK_NOGET.  Vendor-ID and Device-ID should be
applied to hid-core.c, and making an entry to make use of it.

Signed-off-by: Jeffrey Vandenbroucke <jeffrey@wirehead.be>
Cc: Alan Stern <stern@rowland.harvard.edu>
Cc: Greg KH <greg@kroah.com>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/input/hid-core.c |    4 ++++
 1 file changed, 4 insertions(+)

--- gregkh-2.6.orig/drivers/usb/input/hid-core.c
+++ gregkh-2.6/drivers/usb/input/hid-core.c
@@ -1374,6 +1374,9 @@ void hid_close(struct hid_device *hid)
 
 #define USB_VENDOR_ID_PANJIT		0x134c
 
+#define USB_VENDOR_ID_SILVERCREST	0x062a
+#define USB_DEVICE_ID_SILVERCREST_KB	0x0201
+
 /*
  * Initialize all reports
  */
@@ -1680,6 +1683,7 @@ static const struct hid_blacklist {
 	{ USB_VENDOR_ID_HP, USB_DEVICE_ID_HP_USBHUB_KB, HID_QUIRK_NOGET },
 	{ USB_VENDOR_ID_TANGTOP, USB_DEVICE_ID_TANGTOP_USBPS2, HID_QUIRK_NOGET },
 	{ USB_VENDOR_ID_WISEGROUP, USB_DEVICE_ID_DUAL_USB_JOYPAD, HID_QUIRK_NOGET | HID_QUIRK_MULTI_INPUT },
+	{ USB_VENDOR_ID_SILVERCREST, USB_DEVICE_ID_SILVERCREST_KB, HID_QUIRK_NOGET },
 
 	{ USB_VENDOR_ID_APPLE, USB_DEVICE_ID_APPLE_POWERMOUSE, HID_QUIRK_2WHEEL_POWERMOUSE },
 	{ USB_VENDOR_ID_A4TECH, USB_DEVICE_ID_A4TECH_WCP32PU, HID_QUIRK_2WHEEL_MOUSE_HACK_7 },
